hack/deploy.sh

```diff
@@ -1,34 +1,65 @@
-#!/bin/bash
+#!/bin/bash -e
 
-if [ -z "$IMAGE_REPOSITORY" ]; then
-  echo "Not set"
-else
-  echo "$IMAGE_REPOSITORY"
+usage() {
+  cat <<EOF
+Usage: $0
+Environment:
+  IMAGE_URI (required): E.g. quay.io/my_repo/osd-cluster-ready:0.1.38-614bf59
+  JOB_ONLY (optional): If set, only deploy the Job manifest (skip RBAC etc.).
+  DRY_RUN (optional): If set, don't actually deploy anything, just print what would have happened.
+EOF
+  exit -1
+}
+
+maybe() {
+  echo "+ $@"
+  if [[ -z "$DRY_RUN" ]]; then
+    $@
+  fi
+}
+
+if [[ -z "$IMAGE_URI" ]]; then
+  echo "IMAGE_URI not set"
+  usage
 fi
-# Gather commit number for Z and short SHA
-COMMIT_NUMBER=$(git rev-list `git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$"`..HEAD --count)
-CURRENT_COMMIT=$(git rev-parse --short=7 HEAD)
-
-# Build container version
-VERSION_MAJOR=0
-VERSION_MINOR=1
-CONTAINER_VERSION="v$VERSION_MAJOR.$VERSION_MINOR.$COMMIT_NUMBER-$CURRENT_COMMIT"
-
-TMP_MANIFEST=$(mktemp)
-echo "Created $TMP_MANIFEST"
-cat deploy/60-osd-ready.Job.yaml | sed "s/openshift-sre/${IMAGE_REPOSITORY}/" > $TMP_MANIFEST
-sed -i "s/\/osd-cluster-ready/\/osd-cluster-ready:${CONTAINER_VERSION}/" $TMP_MANIFEST
-sed -i "s/value: \"240\"/value: \"339860\"/" $TMP_MANIFEST
+
+TMP_MANIFEST=$(mktemp -t osd-cluster-ready-Job.XXXXX.yaml)
 trap "rm -fr $TMP_MANIFEST" EXIT
+sed "s,\(^ *image: \).*,\1${IMAGE_URI}," deploy/60-osd-ready.Job.yaml > $TMP_MANIFEST
+sed -i 's/value: "240"/value: "339860"/' $TMP_MANIFEST
+echo "===== $TMP_MANIFEST ====="
 cat $TMP_MANIFEST
+echo "========================="
```

Comment:
This was debug, could be removed

Reply:
I like it, especially for dry runs, so I can make sure the `image` came out as I wanted it to with my env vars.